src/Native/Common/pal_config.h.in

```diff
@@ -36,6 +36,9 @@
 #cmakedefine01 HAVE_TCP_VAR_H
 #cmakedefine01 HAVE_RT_MSGHDR
 #cmakedefine01 HAVE_LINUX_RTNETLINK_H
+#cmakedefine01 HAVE_INOTIFY_INIT
```

Comment:
I think it makes more sense to fold these three macros into one that's defined in `configure.cmake`, since we need all three of these APIs. Then all of the `#if HAVE_INOTIFY_X || HAVE_INOTIFY_Y` checks can be folded into something like `#if HAVE_INOTIFY`. We probably also want to make the lack of any of these APIs a compile-time error (or better yet, a configure-time error) on Linux.